src/adaXT/random_forest/RandomForest.py

```diff
@@ -0,0 +1,355 @@
+from multiprocessing import Pool, cpu_count
+from multiprocessing.shared_memory import SharedMemory
+
+from ..decision_tree import DecisionTree
+from ..criteria import Criteria, Squared_error
+from ..decision_tree.splitter import Splitter
+
+
+import numpy as np
+from numpy import float64 as DOUBLE
+import sys
+
+
+class SharedNumpyArray:
+    '''
+    Wraps a numpy array so that it can be shared quickly among processes,
+    avoiding unnecessary copying and (de)serializing.
+    '''
+
+    def __init__(self, array):
+        '''
+        Creates the shared memory and copies the array therein
+        '''
+        # create the shared memory location of the same size of the array
+        self._shared = SharedMemory(create=True, size=array.nbytes)
+
+        # save data type and shape, necessary to read the data correctly
+        self._dtype, self._shape = array.dtype, array.shape
+
+        # create a new numpy array that uses the shared memory we created
+        # at first, it is filled with zeros
+        res = np.ndarray(
+            self._shape, dtype=self._dtype, buffer=self._shared.buf
+        )
+
+        # copy data from the array to the shared memory. numpy will
+        # take care of copying everything in the correct format
+        res[:] = array[:]
+
+    def read(self):
+        '''
+        Reads the array from the shared memory without unnecessary copying.
+        '''
+        # simply create an array of the correct shape and type,
+        # using the shared memory location we created earlier
+        return np.ndarray(self._shape, self._dtype, buffer=self._shared.buf)
+
+    def unlink(self):
+        '''
+        Releases the allocated memory. Call when finished using the data,
+        or when the data was copied somewhere else.
+        '''
+        self._shared.close()
+        self._shared.unlink()
+
+
+class RandomForest:
+    '''
+    The Random Forrest
+    '''
+
+    def __init__(
+            self,
+            forrest_type: str,
+            n_estimators: int = 100,
+            criterion: Criteria = Squared_error,
```

Comment:
I don't think you can just have the default be squared error, no?
What if I am creating a classification tree?

Reply:
Good catch!